src/Tools/a2md/Program.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace a2md
+{
+    public class Program
+    {
+        public static void Main(string[] args)
+        {
+            Loader loader = new Loader();
+            DescriptorEqualityComparer comparer = new DescriptorEqualityComparer();
+
+            var outputMarkdown = args
+                .Select(arg => new AnalyzerFileReference(arg, loader))
+                .SelectMany(analyzerReference => analyzerReference.GetAnalyzersForAllLanguages())
+                .SelectMany(analyzer => analyzer.SupportedDiagnostics)
+                .Distinct(comparer)
```

Comment:
Are we concerned about duplicate ids? It's not the current responsibility of your tool to ensure analyzer consistency but it might be interesting to consider adding some validations along these lines. Preferable might be to do this in context of standard unit tests, though.